### ISSUE TITLE
Can't type some characters

### ISSUE THREAD

**Issue by quantumsheep** [zeit/hyper#3107]
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version

>

- **OS version and name**: Windows 10 Professional v1803 (os version 17134.112)
- **Hyper.app version**: 2.0.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: [QuantumSheep/250cd236a67c9b49703553cea0aa1967](https://gist.github.com/QuantumSheep/250cd236a67c9b49703553cea0aa1967)
- **The issue is reproducible in vanilla Hyper.app**: Yes, just type a '|' or '{' or '[' and it will not be displayed.

## Issue
Some characters can't be typed in Hyper like `|` or `{` or `[` and some others.
It seems that it's only when the `Alt Gr` button is pressed.

**Comment by JoshuaScript** [comment]
This issue seems to only affect international keyboard users, I'm unable to reproduce the issue while using my U.S keyboard layout. Which Layout are you using? I'll try to reproduce the issue.
I also found these similar issues with other terminals:  
 

- https://github.com/jeremyramin/terminal-plus/issues/137
- https://github.com/chjj/term.js/issues/91
- https://teamtreehouse.com/community/how-can-i-type-square-brackets-in-workspaces-on-a-german-keyboard-using-mac

**Comment by QuantumSheep** [comment]
I'm using AZERTY keyboard layout (FR).

**Comment by artero** [comment]
With Spanish keyboards we have problems with the keys ç and ñ

**Comment by harrikauhanen** [comment]
With Finnish keyboard, I cannot enter characters like **£${[]}\@** all requiring *Alt Gr* pressed.

![image](https://user-images.githubusercontent.com/18353/43637716-b1cb6b22-971e-11e8-8638-9215ba89aa6d.png)

**Comment by harrikauhanen** [comment]
For the impatient ones, I created an AutoHotKey script:

```
#NoEnv
SendMode Input
SetWorkingDir %A_ScriptDir%

#if WinActive("ahk_exe Hyper.exe")
  ^>!2::Send {ASC 64}  ; @
  ^>!3::Send {U+00A3}  ; £
  ^>!4::Send {ASC 36}  ; $
  ^>!7::Send {ASC 123} ; {
  ^>!8::Send {ASC 91}  ; [
  ^>!9::Send {ASC 93}  ; ]
#if
```

**Comment by steph-ben** [comment]
> 
> 
> For the impatient ones, I created an AutoHotKey script:
> 
> ```
> #NoEnv
> SendMode Input
> SetWorkingDir %A_ScriptDir%
> 
> #if WinActive("ahk_exe Hyper.exe")
>   ^>!2::Send {ASC 64}  ; @
>   ^>!3::Send {U+00A3}  ; £
>   ^>!4::Send {ASC 36}  ; $
>   ^>!7::Send {ASC 123} ; {
>   ^>!8::Send {ASC 91}  ; [
>   ^>!9::Send {ASC 93}  ; ]
> #if
> ```

Thanks for this script. However, I don't have any clue how to use it ... any tips welcome :-)

**Comment by steph-ben** [comment]
For now this is the reason I cannot use Hyper as my daily terminal. I'm switching back to Cmder

**Comment by nikosalonen** [comment]
Facing same problems with finnish keyboard layout. AltGr + <number> does not work.

![fi-intl](https://user-images.githubusercontent.com/185737/46717363-491f6d80-cc70-11e8-9ae2-887f690d0d59.png)

**Comment by QuantumSheep** [comment]
Soo, I did some research and <kbd>Ctrl</kbd>+<kbd>Alt</kbd> ≈ <kbd>AltGr</kbd>. So Hyper may need to understand when we push <kbd>Alt</kbd> key so it doesn't affect with the other keys.

**Comment by neurotronix** [comment]
Today I have experienced this issue for the first time on Win10 and German keyboard layout. I can perfectly type "{" and "[" anywhere else except in hyper. Weirdly this problem has not been apparent in hyper the last few weeks. Is there any fix?

**Comment by yduman** [comment]
@neurotronix wow indeed! `hyper-pane` caused these issues. Nice catch!

**Comment by chabou** [comment]
Indeed, `hyper-pane` needs better default shortcuts on Windows: https://github.com/chabou/hyper-pane/issues/31

**Comment by djhi** [comment]
I have the same issue on Ubuntu 18.10. I don't use `hyper-pane`. Everything works fine with `hyper_3.0.0-canary.1` but later versions print `AltGr`. Disabling all my plugins does not fix the issue.

I attached my configuration file: [.hyper.js](https://github.com/zeit/hyper/files/2767934/default.hyper.js.txt)

**Comment by gerardbosch** [comment]
I'm trying the version 3.0.2.

Same problem on Spanish layout. It happens to me with the character '~' that is typed with AltGr + 4. The result is the following.

In normal circumstances you would type AltGr + 4, then space --> '~' shows up

What actually happens:
AltGr + 4, then space --> '4~' shows up
AltGr + 4 + 4  --> '44~' shows up

It is very annoying as al need to move the caret back, remove the 4 and move the caret forward.

If I use Ctrl + Alt + 4, then space --> '~' shows up. But this is not practical as I need to move the hands a lot. And is not the standard way :)

@harrikauhannen thanks for the AutoHotkey script, I will try it in the meanwhile.

**Comment by liitfr** [comment]
Hello,
i'm experiencing same trouble.

- windows : Windows 10 Professional, 1803
- hyper : 3.0.2 stable
- keyboard : FR (same key for 2 / é / ~)

on hyper, if i type `alt + 2` then ` `, i get `2~`. Expected result is : `~`

Thanks for your help

**Comment by tcardonne** [comment]
Hi,

I got the same, as commented on #29.

I'm using Windows 10 1903 (1809 at work). Hyper 3.0.2 stable. French keyboard layout.

**Comment by Stanzilla** [comment]
Can you try with a CI build of current canary or maybe compile one yourself?

**Comment by tcardonne** [comment]
Hi @Stanzilla ,

I'm still getting the issue with 3.1.0-canary.1 release (built from cd655ad1, downloaded from appveyor).

**Comment by marcmarina** [comment]
Hello, I still have this issue. I'm on a Spanish layout, and the only Alt Gr combination that doesn't work is Alt Gr + 4 (~).

**Comment by Kevin-De-Koninck** [comment]
I too have the same issue on windows (french azerty keyboard). If I press `altgr+~` I get `=~`.

**Comment by d-weber** [comment]
Bump, same issue here, `<altgr> + <2>` then `<space>` produce `é~`
AZERTY French Keyboard

**Comment by a1ternat1ve** [comment]
Hi, any update? :-) Can't use 1-9 on WSL Ubuntu 22.04, don't have any plugin...
